src/pipeline.hpp

```diff
@@ -21,13 +21,16 @@
 #include <utility>
 #include <vector>
 
-#include "dl_node.hpp"
-#include "entry_node.hpp"
-#include "exit_node.hpp"
 #include "status.hpp"
 
 namespace ovms {
 
+struct Node;
+struct EntryNode;
+struct ExitNode;
+
+using InputPairs = std::vector<std::pair<std::string, std::string>>;
```

Comment:
You call it input pair, but it can also be used to output pair? Maybe StringPair would be better. Or Alias.